File: Binary-Search/2308-divide-array-into-equal-pairs/divide-array-into-equal-pairs.cpp
class Solution {
public:
    bool divideArray(vector<int>& nums) {
        unordered_set<int> check;

        for(int i = 0; i < nums.size(); i++){
            if(check.count(nums[i]) > 0) check.erase(nums[i]);
            else check.insert(nums[i]);
        }

        return check.empty();
    }
};

File: Graphs/2308-divide-array-into-equal-pairs/divide-array-into-equal-pairs.java
class Solution {
    public boolean divideArray(int[] nums) {
        HashSet<Integer> check = new HashSet<>();

        for (int num : nums) {
            if (check.contains(num)) {
                check.remove(num);
            } else {
                check.add(num);
            }
        }

        return check.isEmpty();
    }
}
